src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionSetupForm.php

```diff
@@ -0,0 +1,352 @@
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormInterface $form
+     * @param int $idParentCompanyBusinessUnit
+     *
+     * @return void
+     */
+    protected function updateCompanyBusinessUnitFieldChoices(FormInterface $form, int $idParentCompanyBusinessUnit): void
+    {
+        $form->add(PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_BUSINESS_UNIT, SelectType::class, array_merge(
+            $this->getCompanyBusinessUnitFieldOptions(),
+            [
+                'choices' => $this->getCompanyBusinessUnitChoices($idParentCompanyBusinessUnit),
+            ]
+        ));
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormInterface $form
+     * @param int $idParentCompanyBusinessUnit
+     *
+     * @return void
+     */
+    protected function updateCompanyUserFieldChoices(FormInterface $form, int $idParentCompanyBusinessUnit): void
+    {
+        $form->add(PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_USER, SelectType::class, array_merge(
+            $this->getCompanyUserFieldOptions(),
+            [
+                'choices' => $this->getCompanyUserChoices($idParentCompanyBusinessUnit),
+            ]
+        ));
+    }
+
+    /**
+     * @param int $idParentCompanyBusinessUnit
+     *
+     * @return array
+     */
+    protected function getCompanyBusinessUnitChoices(int $idParentCompanyBusinessUnit): array
+    {
+        $companyBusinessUnitCollectionTransfer = $this->getFactory()
+            ->getCompanyBusinessUnitFacade()
+            ->getCompanyBusinessUnitCollection(
+                (new CompanyBusinessUnitCriteriaFilterTransfer())->setIdParentCompanyBusinessUnit($idParentCompanyBusinessUnit)
+            );
+
+        $parentCompanyBusinessUnitTransfer = $this->getFactory()
+            ->getCompanyBusinessUnitFacade()
+            ->findCompanyBusinessUnitById($idParentCompanyBusinessUnit);
+
+        if (!$parentCompanyBusinessUnitTransfer) {
+            return [];
+        }
+
+        $companyBusinessUnits = [];
+        $companyBusinessUnitTransfers = $companyBusinessUnitCollectionTransfer->getCompanyBusinessUnits()->count()
+            ? $companyBusinessUnitCollectionTransfer->getCompanyBusinessUnits()
+            : [$parentCompanyBusinessUnitTransfer];
+
+        foreach ($companyBusinessUnitTransfers as $companyBusinessUnitTransfer) {
+            $companyBusinessUnits[$companyBusinessUnitTransfer->getName()] = $companyBusinessUnitTransfer->getIdCompanyBusinessUnit();
+        }
+
+        return $companyBusinessUnits;
+    }
+
+    /**
+     * @param int $idParentCompanyBusinessUnit
+     *
+     * @return array
+     */
+    protected function getCompanyUserChoices(int $idParentCompanyBusinessUnit): array
+    {
+        $companyUserCollectionTransfer = $this->getFactory()
+            ->getCompanyUserFacade()
+            ->getCompanyUserCollection(
+                (new CompanyUserCriteriaFilterTransfer())->setIdCompanyBusinessUnit($idParentCompanyBusinessUnit)
```

Comment:
Please move `->setIdCompanyBusinessUnit($idParentCompanyBusinessUnit)` to the next line.